src/components/site/header/HeaderProgress.jsx

```diff
@@ -1,40 +1,63 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import last from 'lodash/last';
 import CustomIcon from '../icons/CustomIcon';
 import LinearProgress from '../linearProgress';
 import LinearProgressMarker from '../linearProgress/LinearProgressMarker';
 
 class HeaderProgress extends React.PureComponent {
+  getDisplayValue = (currentPage, totalPages) =>
+    currentPage ? Math.round((currentPage / totalPages) * 100) : 0;
+
+  getProgressValue = (currentPage, section) => {
+    const firstSectionPage = section[0];
+    const lastSectionPage = last(section);
+    const sectionRange = lastSectionPage - firstSectionPage;
+
+    if (currentPage < firstSectionPage) return null;
+    if (currentPage >= lastSectionPage) return 100;
+
+    return Math.round(((currentPage - firstSectionPage) / sectionRange) * 100);
+  };
+
   render = () => {
-    const { checkpoints, pageNumber, totalPages } = this.props;
+    const { pageNumber, totalPages, sections } = this.props;
+
     return (
       <div className="header-progress-wrapper">
-        <LinearProgress
-          id="current-page-of-total"
-          value={pageNumber ? Math.round((pageNumber / totalPages) * 100) : 0}
-        >
-          {checkpoints && (
-            <div className="header-progress-checkpoints">
-              {checkpoints.map(checkpoint => {
-                return (
+        {sections &&
+          sections.map((section, i) => {
+            const key = `section-${i}`;
+            const lastPage = last(section);
+
+            return (
+              <LinearProgress
+                key={key}
+                value={this.getProgressValue(pageNumber, section)}
+                displayValue={this.getDisplayValue(pageNumber, totalPages)}
+                style={{ width: `${(section.length / totalPages) * 100}%` }}
+              >
+                {lastPage !== totalPages && (
+                  <LinearProgressMarker
+                    filled
+                    completed={pageNumber >= lastPage}
+                    progress={100}
+                    style={{ zIndex: pageNumber === lastPage ? -1 : 1 }}
```

Comment:
set the z-index if the progress marker overlaps with the ending marker so that the progress marker displays on top. The progress marker cannot default display on top because after a progress bar is completed the marker is left at 100% but hidden by the ending marker.